Controller/TreeController.php

```diff
@@ -84,4 +102,48 @@ public function treeAction(Request $request)
             //'delete_in_overlay' => $deleteInOverlay ? $deleteInOverlay : false,
         ));
     }
+
+    /**
+     * Reorder $moved (child of $parent) before or after $target.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function reorderAction(Request $request)
+    {
+        $parentPath = $request->get('parent');
+        $dropedAtPath = $request->get('dropped');
+        $targetPath = $request->get('target');
+        $position = $request->get('position');
+
+        if (null === $parentPath || null === $dropedAtPath || null === $targetPath) {
+            return new JsonResponse(['Parameters parent, dropped and target has to be set to reorder.'], Response::HTTP_BAD_REQUEST);
+        }
+
+        $before = 'before' == $position;
+        $parentNode = $this->session->getNode($parentPath);
+        $targetName = PathHelper::getNodeName($targetPath);
+        if (!$before) {
+            $nodesIterator = $parentNode->getNodes();
+            $nodesIterator->rewind();
+            while ($nodesIterator->valid()) {
+                if ($nodesIterator->key() == $targetName) {
+                    break;
+                }
+                $nodesIterator->next();
+            }
+            $targetName = null;
+            if ($nodesIterator->valid()) {
+                $nodesIterator->next();
+                if ($nodesIterator->valid()) {
+                    $targetName = $nodesIterator->key();
+                }
+            }
+        }
+        $parentNode->orderBefore($targetName, PathHelper::getNodeName($dropedAtPath));
```

Comment:
@dbu  this one was in the wrong direction on `1.x` (where i copied the code), so that can't have ever worked.